Symfony 6 upgrade + other maintenance

MKodde opened the pull request:
1. GHA was put into place, Travis was laid to rest.
  1. For test-integration
  2. For nightly sec-checks
2. Ant task runner was replaced  by Composer scripts to run the QA tests
3. Rector was used to update /src including the tests
4. By adding type safety, some tests were removed that tested input validation

MKodde committed:
Install Composer upgrades

Require SF 5 or 6
Require PHP 8.1+

MKodde committed:
Make tests PHPUnit 10 compatible

MKodde committed:
Replace Ant test runner with Composer scripts

MKodde committed:
Update code to be code style complient

MKodde committed:
Update .gitignore to exclude .phpunit.result.cache

MKodde committed:
Make code SF6 & PHP8 friendly

MKodde committed:
Replace Travis with GitHub Actions

For test-integration and nightly security checks

MKodde committed:
Update CHANGELOG

MKodde committed:
Upgrade Guzzle to v7

MKodde committed:
Upgrade Monolog to v3

MKodde committed:
Remove Framework Bundle

MKodde committed:
Update PHPMD config to exclude tests

MKodde committed:
Convert JsonConvertibleParamConverter to a Resolver

MKodde committed:
Install PHPStan and configure it

MKodde committed:
Move MissingRequiredAttributeException to stepup bundle

MKodde committed:
Repair phpstan findings

Settled at level 5 with minial basline additions

MKodde committed:
Allow tests on 8.3 to fail for now

MKodde committed:
Add update baseline script and raise baseline

MKodde committed:
Uninstall rector/rector

MKodde commented on src/Resources/config/services.yml:
```diff
@@ -7,14 +7,12 @@ services:
 
-    surfnet_stepup.json_convertible_param_converter:
-        class: Surfnet\StepupBundle\Request\JsonConvertibleParamConverter
+    Surfnet\StepupBundle\Request\JsonConvertibleResolver:
+        class: Surfnet\StepupBundle\Request\JsonConvertibleResolver
```
Is this not a default framework feature? Is this still required?

MKodde reviewed (commented):
The following feedback points came up during a peer review session with @pablothedude.

Thanks for you input!

MKodde committed:
Stop using `session` in SessionSmsVerificationStateHandler

MKodde committed:
Refrain from using twig error controller

tvdijen commented on composer.json:
```diff
@@ -14,30 +14,30 @@
-        "squizlabs/php_codesniffer": "^3.4",
-        "symfony/phpunit-bridge": "^4.4|^5.4"
+        "mockery/mockery": "^1.5",
+        "overtrue/phplint": "*",
```
What would be the pros/cons of the SensioLabs sniffer over the one from SquizLabs?

parijke committed:
Changed array to LogRecord object

Changed array to LogRecord object

parijke committed:
Changed array to LogRecord object

parijke committed:
Changed array to LogRecord object

parijke committed:
Changed array to LogRecord object

MKodde committed:
Raise minimal PHP version constraint to 8.2

MKodde committed:
Do not wire the optional surfnet_stepup.service.challenge_handler

MKodde committed:
Remove the custom Exception Listener

The default behaviour now shipping with SF6 no longer requires us to
provide our own listener

MKodde committed:
Update PHPSTan baseline

MKodde committed:
Make Art aware of Throwable errors and exceptions

MKodde committed:
Improve JsonConvertibleResolver

The resolver now checks if it is appropriate to try'n resolve the input.

This solves the Syntax Error issue that would pop up when a parameter
was converted that was completely unsuitable to convert.

MKodde committed:
Allow for extensions to access the fields

MKodde committed:
Provide test double for SmsRecoveryTokenService

MKodde committed:
Introduce the SendSmsChallengeCommandInterface

And the VerifyPossessionOfPhoneCommandInterface. They are now used in
the SmsRecoveryTokenService implementations

MKodde committed:
Address deprecation warnings

parijke committed:
Update ViewConfigCollection.php

Add nullable to the type

MKodde committed:
Improve LoaResolutionService type definitions

MKodde committed:
Let JsonConvertibleResolver yield instead of return

That fixed the ::resolve()\" must yield at most one value for non-variadic arguments.\" at ArgumentResolver.php
issue.

parijke committed:
Update LoaResolutionService.php

Type int is incorrect, since we also use Loa's like 1.5.

parijke committed:
Update SmsRecoveryTokenServiceInterface.php

Strict types spacing

parijke committed:
Set unit test earlier in the check chain

parijke committed:
Fix unit test using generators

The resolve method returns a generator. This is postponed, until the php code is actually using is to preserve memory. Therfor, calling ->currnt() on it forces the method to be executed

parijke committed:
Change translation variable using {} instead of %%

See https://symfony.com/doc/current/reference/formats/message_format.html for details

parijke committed:
Revert "Change translation variable using {} instead of %%"

This reverts commit 2f2e086f3f9ce732f419f59512bfa35abc3a2811.
